hadoop-project/pom.xml

```diff
@@ -167,7 +167,7 @@
 
     <!-- Plugin versions and config -->
     <maven-surefire-plugin.argLine>-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError</maven-surefire-plugin.argLine>
-    <maven-surefire-plugin.version>3.0.0-M1</maven-surefire-plugin.version>
+    <maven-surefire-plugin.version>3.0.0-M7</maven-surefire-plugin.version>
```

Comment:
This needed? Intentional?

Reply:
This is needed because the Surefire plugging 3.0.0-M1 wasn't including the launcher as part of it's setup, so the Yarn tests were failing.  Switching to 3.0.0-M7 fixes that issue, and demonstrates that the parallel build operate correctly.

Reply:
ok. should we make this a separate PR so it stands out and can be backported more easily?

Reply:
I've cherry-picked just the upgrade to 3.0.0-M7.
https://github.com/apache/hadoop/pull/4795